javaparser-core-generators/src/main/java/com/github/javaparser/generator/core/visitor/HashCodeVisitorGenerator.java

```diff
@@ -22,19 +22,29 @@ public HashCodeVisitorGenerator(SourceRoot sourceRoot) {
     }
 
     @Override
-    protected void generateVisitMethodBody(BaseNodeMetaModel node, MethodDeclaration visitMethod, CompilationUnit compilationUnit) {
+    protected void generateVisitMethodBody(BaseNodeMetaModel node, MethodDeclaration visitMethod,
+                                           CompilationUnit compilationUnit) {
         visitMethod.getParameters().forEach(p -> p.setFinal(true));
 
         final BlockStmt body = visitMethod.getBody().get();
         body.getStatements().clear();
 
         final SeparatedItemStringBuilder builder = new SeparatedItemStringBuilder("return ", "* 31 +", ";");
-        final List<PropertyMetaModel> propertyMetaModels= node.getAllPropertyMetaModels();
-        if (propertyMetaModels.isEmpty()) {
+        final List<PropertyMetaModel> propertyMetaModels = node.getAllPropertyMetaModels();
+        String ndName = node.getTypeName();
+        if (ndName.equals("JavadocComment") || ndName.equals("LineComment") || ndName.equals("BlockComment")
+                || propertyMetaModels.isEmpty()) {
```

Comment:
This can be done in a nicer way: there is exactly one meta model instance for every node and property, so you can directly compare `node` to one of the fields in `JavaParserMetaModel`. (I'm writing this without checking, so if it turns out it makes no sense I'll see where I went wrong ;-) )

Reply:
@matozoid: I was hoping you brought this up. If hypothetically, the class name of one those nodes changed, this code would break. Thanks!

Reply:
No problem. If you ever think "hey, I should use reflection on the nodes to do x" - don't do it, use the meta model :-)